src/pkgdb/pkg-query.cc

```diff
@@ -226,41 +237,69 @@ PkgQuery::initMatch()
 {
   if ( this->match.has_value() && ( ! this->match->empty() ) )
     {
-      this->addWhere(
-        "( pname LIKE :match ) OR ( description LIKE :match )"
-      );
+      /* If `match' is set `matchStyle' must be set too. */
+      switch( this->matchStyle )
+        {
+
+          case QMS_RESOLVE:
+            this->addSelection( "( :match = pname ) AS matchExactPname" );
```

Comment:
I think it might be more clear if this was a distinct field from match. I'm not sure but I wonder if that might even allow dropping the `case` - maybe rather than special casing for resolve vs search, there can just be different knobs (e.g. exactName vs match), and resolve vs search can decide which knobs to use.

Reply:
I had considered it, and it has seemed appealing throughout working on this; but because we really rely on _nearly identical_ behavior for search and resolution. I want to keep a single routine to run all filtering, ordering, etc.

My concern is that straying from DRY here could quickly cause drift and become a maintenance headache. It's easy to forget to copy a block from one a search query implementation over to the resolution query implementation.

As for whether `match` should be one field or two different ones you might be right.
We can pair after standup maybe and refactor that part together if you'd like.